include/RED4ext/Containers/DynArray.hpp

```diff
@@ -0,0 +1,518 @@
+    {
+        return capacity;
+    }
+
+    [[nodiscard]] constexpr SizeType MaxSize() const noexcept
+    {
+#undef max // preprocessor thinks max is a macro under NOMINMAX
+        return std::numeric_limits<SizeType>::max();
+    }
+
+    [[nodiscard]] constexpr SizeType Size() const noexcept
+    {
+        return size;
+    }
+
+#pragma region STL
+#pragma region Iterator
+    [[nodiscard]] constexpr Iterator begin() noexcept
+    {
+        return Begin();
+    }
+
+    [[nodiscard]] constexpr ConstIterator begin() const noexcept
+    {
+        return Begin();
+    }
+
+    [[nodiscard]] constexpr ConstIterator cbegin() const noexcept
+    {
+        return begin();
+    }
+
+    [[nodiscard]] constexpr Iterator end() noexcept
+    {
+        return End();
+    }
+
+    [[nodiscard]] constexpr ConstIterator end() const noexcept
+    {
+        return End();
+    }
+
+    [[nodiscard]] constexpr ConstIterator cend() const noexcept
+    {
+        return end();
+    }
+#pragma endregion
+#pragma region Reverse Iterator
+    [[nodiscard]] constexpr ReverseIterator rbegin() noexcept
+    {
+        return RBegin();
+    }
+
+    [[nodiscard]] constexpr ConstReverseIterator rbegin() const noexcept
+    {
+        return RBegin();
+    }
+
+    [[nodiscard]] constexpr ConstReverseIterator crbegin() const
+    {
+        return rbegin();
+    }
+
+    [[nodiscard]] constexpr ReverseIterator rend() noexcept
+    {
+        return REnd();
+    }
+
+    [[nodiscard]] constexpr ConstReverseIterator rend() const noexcept
+    {
+        return REnd();
+    }
+
+    [[nodiscard]] constexpr ConstReverseIterator crend() const noexcept
+    {
+        return rend();
+    }
+#pragma endregion
+#pragma endregion
+private:
+    T* entries;        // 00
+    uint32_t capacity; // 08
+    uint32_t size;     // 0C
+
+    void MoveEntries(Pointer aSrc, Pointer aDst, SizeType aCount)
+    {
+        if (aCount == 0 || aSrc == aDst)
+            return;
+
+        if (aSrc < aDst)
+        {
+            std::move_backward(aSrc, aSrc + aCount, aDst + aCount);
+        }
+        else
+        {
+            std::move(aSrc, aSrc + aCount, aDst);
+        }
+    }
+
+    SizeType CalculateGrowth(SizeType aNewSize)
+    {
+        uint32_t geometric = Capacity() + (Capacity() / 2);
+        return (std::max)(aNewSize, geometric);
+    }
+
+    void SetCapacity(SizeType aNewCapacity)
+    {
+        if (aNewCapacity < size)
+            return;
+
+        constexpr uint32_t alignment = alignof(ValueType);
+
+        using func_t =
+            void (*)(DynArray* aThis, uint32_t aCapacity, uint32_t aElementSize, uint32_t aAlignment,
+                     void (*aMoveFunc)(Pointer aDstBuffer, Pointer aSrcBuffer, int32_t aSrcSize, DynArray* aSrcArray));
+
+        static UniversalRelocFunc<func_t> func(Detail::AddressHashes::DynArray_Realloc);
+        func(this, aNewCapacity, sizeof(ValueType), alignment >= 8 ? alignment : 8, nullptr);
```

Comment:
The last argument must be a pointer to function that moves elements to a new buffer. When `nullptr` is passed, it just copies memory to a new location.